app/models/user.rb

```diff
@@ -33,11 +33,79 @@ class User < ApplicationRecord
 
   scope :remote_lawyers, -> { where(remote_clinic_lawyer: true) }
 
+  filterrific(
+    default_filter_params: { sorted_by: 'created_at_desc' },
+    available_filters: %i[
+      filter_first_name
+      filter_last_name
+      filter_email
+      with_volunteer_type
+      sorted_by
+
+    ]
+  )
+
+  def self.options_for_sorted_by
+    [
+      ['Created at (newest first)', 'created_at_desc'],
+      ['Created at (oldest first)', 'created_at_asc']
+    ]
+  end
+
   def remote_clinic_friends
     friends.where(user_friend_associations: { remote: true })
   end
 
-  scope :for_volunteer_type, ->(volunteer_type) { where(volunteer_type: volunteer_type) }
+  scope :sorted_by, ->(sort_option) {
+    direction = sort_option =~ /desc$/ ? 'desc' : 'asc'
+    case sort_option.to_s
+    when /^created_at_/
+      order("users.created_at #{direction}")
+    when /^first_name_/
+      order("LOWER(users.first_name) #{ direction }")
+    when /^last_name_/
+      order("LOWER(users.last_name) #{ direction }")
+    when /^email_/
+      order("users.email #{direction}")
+    when /^admin?_/
+      order("users.admin? #{direction}")
+    else
+      raise(ArgumentError, "Invalid sort option: #{sort_option.inspect}")
+    end
+  }
+
+  scope :search_query, lambda { |query|
```

Comment:
Is this being used? 

If so - it might be a bit easier to read if you go ahead and make it a block like: 
```
scope :search_query, lambda do |query|
    ...
end
```

Reply:
Oof. yes, that is hard to read. I'm not even sure I put the do and end in the right places. Will you do a sanity check on this edit for me? Much obliged!!!